src/mender-update/context/context.cpp

```diff
@@ -283,6 +276,29 @@ static error::Error FilterProvides(
 	return error::NoError;
 }
 
+// Used in testing
+expected::ExpectedBool ProvidesCheckClearProvides(
+	ProvidesData &provides,
+	const ProvidesData &new_provides,
+	const ClearsProvidesData &clears_provides) {
+	auto err = FilterProvides(new_provides, clears_provides, provides);
+	if (err != error::NoError) {
+		return expected::unexpected(err);
+	}
+
+	set<string> clears;
+	for (const auto &clear : clears_provides) {
+		clears.insert(clear);
+	}
+
+	for (const auto &provide : provides) {
+		if (clears.find(provide.first) != clears.end()) {
+			return false;
+		}
+	}
+	return true;
```

Comment:
I don't quite follow why this function is here. Why not use the original directly?

Also, this function returns both `false` and `unexpected` in certain cases, but none of these conditions are being tested. At least the function should live in the test file then, if the false/error conditions are only meant to catch mistakes.

Reply:
Well, `FilterProvides` is a static function, and I implemented the tests based on the test for `FilterProvides` in the orchestrator. I now realize, however, that the function doesn't work as intended anyways. So either I have to change the function to make it actually work or make `FilterProvides` accessible for the tests. What do you think?

Reply:
It's fine to make the function non-static and add it to the header file. This is quite common to do in test situations such as this one, and is usually preferred over adding "special" test functions. The closer to the real thing you are, the better! 🙂